### ISSUE TITLE
`prefer-simple-condition-first`: Avoid reporting `typeof` guards for possibly undeclared globals

### ISSUE THREAD

**Issue by Sli4o** [sindresorhus/eslint-plugin-unicorn#3732]
### Description

`prefer-simple-condition-first` currently reports logical expressions where a `typeof` check intentionally guards access to a global that may not exist at runtime.

For TypeScript declarations such as `declare const`, the identifier is known to the type system, but no runtime binding is emitted. This means the order of the conditions is semantically significant.

This is related to #3701, which addressed the same distinction for `no-typeof-undefined`.

### Example

```ts
declare const COMPILE_TIME_FLAG: boolean | undefined;

if (typeof COMPILE_TIME_FLAG !== 'undefined' && COMPILE_TIME_FLAG) {
    console.log('Compiled');
}
```

`prefer-simple-condition-first` reports `COMPILE_TIME_FLAG` and suggests:

> Consider moving this simple condition first after verifying short-circuit behavior.

However, moving it first is not safe:

```ts
if (COMPILE_TIME_FLAG && typeof COMPILE_TIME_FLAG !== 'undefined') {
    console.log('Compiled');
}
```

If `COMPILE_TIME_FLAG` was not injected by the build system, evaluating it directly throws a `ReferenceError`.

The original order is intentionally safe because `typeof` can be used with an undeclared identifier:

```ts
typeof COMPILE_TIME_FLAG !== 'undefined' && COMPILE_TIME_FLAG
```

### Expected behavior

`prefer-simple-condition-first` should not report a condition when an earlier `typeof identifier !== 'undefined'` check guards a later direct access to the same identifier.

In particular, this should remain valid:

```ts
typeof COMPILE_TIME_FLAG !== 'undefined' && COMPILE_TIME_FLAG
```

because reordering the operands changes runtime behavior.

This case is especially relevant for compile-time defines provided by bundlers or compilers and declared to TypeScript using `declare const`.

Related: #3701.
